app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModel.kt

```diff
@@ -41,7 +43,7 @@ class TransactionDetailsViewModel
             executionInfo.signers.contains(credentials.address) && !executionInfo.confirmations.map { it.signer }.contains(credentials.address)
         }
 
-    fun submitConfirmation(transaction: TransactionDetails, executionInfo: DetailedExecutionInfo.MultisigExecutionDetails) {
+    fun submitConfirmation(transaction: TransactionDetailsViewData, executionInfo: DetailedExecutionInfo.MultisigExecutionDetails) {
```

Comment:
actually there is no need to pass any parameter here. We could hold TxDetails model internally in view model and only expose view data to fragment. This way we won't need to move calculateTxHash from :data to :app